### PR TITLE
Cache Leiningen and Boot dependencies

### PR DESCRIPTION
**Pull request by adamrenklint**
This PR introduces caching of Leiningen and Boot dependencies, i.e. `~/.m2` and `~/.boot`, making Netlify a great tool to build static sites and SPAs with Clojure and ClojureScript.

### CODE REVIEW

**Comment by rybit** [comment]
Sorry for the delay on this! We will take a chance to review this soon, but the holidays are just a bit slow (everyone is off!)

**Comment by adamrenklint** [comment]
Please let me know if you want any changes to this, to get it merged :)

File: run-build-functions.sh [hunk 1]
```diff
@@ -306,11 +308,14 @@ install_dependencies() {
   # Leiningen
   if [ -f project.clj ]
   then
-    mkdir -p $NETLIFY_CACHE_DIR/m2
-    ln -nfs $NETLIFY_CACHE_DIR/m2 ~/.m2
+    if [ -d $NETLIFY_CACHE_DIR/.m2 ];
```
Comment (brycekahle) [inline run-build-functions.sh]: Please remove the semicolon. It looks like we aren't 100% consistent, but the vast majority do not have semicolons.

File: run-build-functions.sh [hunk 2]
```diff
@@ -327,8 +332,16 @@ install_dependencies() {
   # Boot
   if [ -f build.boot ]
   then
-    mkdir -p $NETLIFY_CACHE_DIR/m2
-    ln -nfs $NETLIFY_CACHE_DIR/m2 ~/.m2
+    if [ -d $NETLIFY_CACHE_DIR/.m2 ];
```
Comment (brycekahle) [inline run-build-functions.sh]: Remove semicolon

File: run-build-functions.sh [hunk 3]
```diff
@@ -327,8 +332,16 @@ install_dependencies() {
...
+    then
+      rm -rf $NETLIFY_BUILD_BASE/.m2
+      mv $NETLIFY_CACHE_DIR/.m2 $NETLIFY_BUILD_BASE/.m2
+    fi
+    if [ -d $NETLIFY_CACHE_DIR/.boot ];
```
Comment (brycekahle) [inline run-build-functions.sh]: Remove semicolon

**Comment by brycekahle** [comment]
@adamrenklint This won't go out until Monday most likely. I'll be sure to ping you when it does.

**Pull request merged**